### PR TITLE
[DO NOT MERGE] Document Micrometer metrics

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
❗ DO NOT MERGE BEFORE 0.10.0 RELEASE ❗

### CODE REVIEW

File: docs/metrics.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# Metrics
+
+> :construction: This feature is experimental - exported metric data and configuration properties may change.
+
+The Splunk distribution of OpenTelemetry Java agent is also gathering basic application metrics. We
```
Comment (iNikem) [inline docs/metrics.md:5]:
```suggestion
The Splunk Distribution of OpenTelemetry Java agent gathers basic application metrics. We
```

File: docs/metrics.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+to gather and export metrics to either [SignalFx SmartAgent](https://github.com/signalfx/signalfx-agent/)
+or the [Splunk distribution of OpenTelemetry Collector](https://github.com/signalfx/splunk-otel-collector).
+
+## Default metric tags
+
+The following dimensions are automatically added to all metrics exported by the agent:
+
+| Tag name                 | Tag value |
+| ------------------------ | --------- |
+| `deployment.environment` | The value of the `deployment.environment` resource attribute.
```
Comment (iNikem) [inline docs/metrics.md:17]: Should we add "if present"?

File: docs/metrics.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+| Metric name                    | Description |
+| ------------------------------ | ----------- |
+| `db.pool.connections`          | The number of open connections.
+| `db.pool.connections.active`   | The number of open connections that are currently in use.
+| `db.pool.connections.idle`     | The number of open connections that are currently idle.
+| `db.pool.connections.idle.min` | The minimum number of idle open connections allowed.
+| `db.pool.connections.idle.max` | The maximum number of idle open connections allowed.
+| `db.pool.connections.max`      | The maximum number of open connections allowed.
+
+All `apache-dbcp2` metrics have the following tags:
```
Comment (iNikem) [inline docs/metrics.md:87]: This `apache-dbcp2` looks like id or something similarly significant (as opposed to just "memory pools" above). Is it?

Comment (mateuszrzeszutek) [reply docs/metrics.md:87]: No, that's just the name of the library. I'll use "Apache DBCP2" instead, same as in the header.

**Review by iNikem — APPROVED** [review]

**Pull request merged**